macosx/FilterBarController.mm

```diff
@@ -266,7 +266,9 @@ - (void)focusSearchField
 
 - (BOOL)isFocused
 {
-    return self.fSearchField.window.firstResponder == self.fSearchField;
+    return ([[[self.fSearchField window] firstResponder] isKindOfClass:[NSTextView class]] &&
+            [[self.fSearchField window] fieldEditor:NO forObject:nil] != nil &&
+            [self.fSearchField isEqualTo:(id)[(NSTextView *)[[self.fSearchField window] firstResponder]delegate]]);
```

Comment:
nit:
- Prefer a space between `]` and `delegate` for consistency with all the objective-c code in the project.
- Prefer avoiding casting to `(id)`.
- Superfluous parenthesis.

ie:
```
    return [[[self.fSearchField window] firstResponder] isKindOfClass:[NSTextView class]] &&
        [[self.fSearchField window] fieldEditor:NO forObject:nil] != nil &&
        [self.fSearchField isEqualTo:[(NSTextView*)[[self.fSearchField window] firstResponder] delegate]];
```

Reply:
Thanks for review. Fixed it.

Reply:
Do I need to fix code style again because [code-style](https://github.com/transmission/transmission/actions/runs/3559715798/jobs/5979254141) check didn’t pass.

Reply:
I pushed an update for code-style, so no worries